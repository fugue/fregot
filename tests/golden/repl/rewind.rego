package rewind

crashing_rule {
  x = 1
  x + input.msg with input as {"msg": "crash"}
}

failing_rule {
  x = 1
  x = 2
}
