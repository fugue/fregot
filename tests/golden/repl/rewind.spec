{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file", "rewind.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"stdout": "${SPEC_NAME}.stdout"}
    ],
    "stdin": [
        ":break rewind.crashing_rule",
        "data.rewind.crashing_rule",
        ":next",
        ":next",
        ":rewind",
        ":quit",
        ":break rewind.failing_rule",
        "data.rewind.failing_rule",
        ":next",
        ":next",
        ":rewind",
        ":quit"
    ]
}
