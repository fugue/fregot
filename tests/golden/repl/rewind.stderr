F u g u e   R E G O   T o o l k i t
fregot v0.2.0 repl - use :help for usage info
fregot ([34meval error[0m):
  "rewind.rego" (line 5, column 3):
  builtin type error:

    5|   x + "hi" > "wat"
         [31m^^^^^^^^[0m

  Expected number but got string

  Stack trace:
    rule [33mrewind[33m.[0mcrashing_rule[0m at data.rewind.crashing_rule:1:1
